### ISSUE TITLE
QGIS crashes when loading GPX file

### ISSUE THREAD

**Issue by dhdeangelis** [qgis/QGIS#55558]
### What is the bug or the crash?

I compiled today's (2023-12-07) master without problems. All older projects load and work correctly except one that has a lot of GPX files. When opening this project QGIS crashes. Further investigation revealed that loading a GPX file into a new project will also make QGIS crash. Tested different GPX files and QGIS crashes the same.

Trace loading project with GPX files:
```
> qgis
Warning 1: GPX schema version is unknown. The driver may not be able to handle the file correctly and will behave as if it is GPX 1.1.
Warning 1: GPX schema version is unknown. The driver may not be able to handle the file correctly and will behave as if it is GPX 1.1.
QGIS died on signal 11[New LWP 4213]
[New LWP 4214]
[New LWP 4221]
[New LWP 4222]
[New LWP 4223]
[New LWP 4224]
[New LWP 4226]
[New LWP 4227]
[New LWP 4228]
[New LWP 4229]
[New LWP 4230]
[New LWP 4231]
[New LWP 4232]
[New LWP 4233]
[New LWP 4234]
[New LWP 4235]
[New LWP 4236]
[New LWP 4237]
[New LWP 4238]
[New LWP 4239]
[New LWP 4240]
[New LWP 4241]
[New LWP 4242]
[New LWP 4243]
[New LWP 4244]
[New LWP 4245]
[New LWP 4246]
[New LWP 4247]

This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.opensuse.org/>
Enable debuginfod for this session? (y or [n]) [answered N; input not from terminal]
Debuginfod has been disabled.
To make this setting permanent, add 'set debuginfod enabled off' to .gdbinit.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
0x00007f586eb09d7f in poll () from /lib64/libc.so.6
[Current thread is 1 (Thread 0x7f586c416680 (LWP 4212))]
#0  0x00007f586eb09d7f in poll () at /lib64/libc.so.6
#1  0x00007f586e515aff in  () at /lib64/libglib-2.0.so.0
#2  0x00007f586e51620c in g_main_context_iteration () at /lib64/libglib-2.0.so.0
#3  0x00007f5878d460d6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt5Core.so.5
#4  0x00007f5878cebc2b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt5Core.so.5
#5  0x00007f5878cf40c0 in QCoreApplication::exec() () at /lib64/libQt5Core.so.5
#6  0x000000000041554c in main ()
[Inferior 1 (process 4212) detached]
Warning 1: GPX schema version is unknown. The driver may not be able to handle the file correctly and will behave as if it is GPX 1.1.
QGIS died on signal 11gdb returned 0
Aborted (core dumped)
hernan@localhost:~> ptrace: No such process.
No thread selected
No stack.
```

Trace loading a GPX file into a new project:
```
>qgis
QGIS died on signal 11[New LWP 4320]
[New LWP 4321]
[New LWP 4328]
[New LWP 4329]
[New LWP 4330]
[New LWP 4333]
[New LWP 4334]
[New LWP 4335]
[New LWP 4336]
[New LWP 4337]
[New LWP 4338]
[New LWP 4339]
[New LWP 4340]
[New LWP 4341]
[New LWP 4342]
[New LWP 4343]
[New LWP 4344]
[New LWP 4345]
[New LWP 4346]
[New LWP 4347]
[New LWP 4348]
[New LWP 4349]
[New LWP 4350]
[New LWP 4351]
[New LWP 4352]
[New LWP 4353]
[New LWP 4354]
[New LWP 4358]
[New LWP 4359]
[New LWP 4367]

This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.opensuse.org/>
Enable debuginfod for this session? (y or [n]) [answered N; input not from terminal]
Debuginfod has been disabled.
To make this setting permanent, add 'set debuginfod enabled off' to .gdbinit.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
0x00007f1a69f09d7f in poll () from /lib64/libc.so.6
[Current thread is 1 (Thread 0x7f1a67b1b680 (LWP 4319))]
#0  0x00007f1a69f09d7f in poll () at /lib64/libc.so.6
#1  0x00007f1a69915aff in  () at /lib64/libglib-2.0.so.0
#2  0x00007f1a6991620c in g_main_context_iteration () at /lib64/libglib-2.0.so.0
#3  0x00007f1a73f460d6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt5Core.so.5
#4  0x00007f1a73eebc2b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt5Core.so.5
#5  0x00007f1a74dadc67 in QDialog::exec() () at /lib64/libQt5Widgets.so.5
#6  0x00007f1a77ebddf9 in QgsAppLayerHandling::addOgrVectorLayers(QStringList const&, QString const&, QString const&, bool&, bool) () at /usr/local/lib/libqgis_app.so.3.35.0
#7  0x00007f1a779c3c04 in QgisApp::dataSourceManager(QString const&, QString const&)::{lambda(QStringList const&, QString const&, QString const&)#1}::operator()(QStringList const&, QString const&, QString const&) const () at /usr/local/lib/libqgis_app.so.3.35.0
#8  0x00007f1a77a5b846 in QtPrivate::FunctorCall<QtPrivate::IndexesList<0, 1, 2>, QtPrivate::List<QStringList const&, QString const&, QString const&>, void, QgisApp::dataSourceManager(QString const&, QString const&)::{lambda(QStringList const&, QString const&, QString const&)#1}>::call(QgisApp::dataSourceManager(QString const&, QString const&)::{lambda(QStringList const&, QString const&, QString const&)#1}&, void**) () at /usr/local/lib/libqgis_app.so.3.35.0
#9  0x00007f1a77a5900b in void QtPrivate::Functor<QgisApp::dataSourceManager(QString const&, QString const&)::{lambda(QStringList const&, QString const&, QString const&)#1}, 3>::call<QtPrivate::List<QStringList const&, QString const&, QString const&>, void>(QgisApp::dataSourceManager(QString const&, QString const&)::{lambda(QStringList const&, QString const&, QString const&)#1}&, void*, void**) () at /usr/local/lib/libqgis_app.so.3.35.0
#10 0x00007f1a77a53b00 in QtPrivate::QFunctorSlotObject<QgisApp::dataSourceManager(QString const&, QString const&)::{lambda(QStringList const&, QString const&, QString const&)#1}, 3, QtPrivate::List<QStringList const&, QString const&, QString const&>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) () at /usr/local/lib/libqgis_app.so.3.35.0
#11 0x00007f1a73f25442 in  () at /lib64/libQt5Core.so.5
#12 0x00007f1a75c40e87 in QgsDataSourceManagerDialog::addVectorLayers(QStringList const&, QString const&, QString const&) () at /usr/local/lib/libqgis_gui.so.3.35.0
#13 0x00007f1a76b199e0 in QgsDataSourceManagerDialog::vectorLayersAdded(QStringList const&, QString const&, QString const&) () at /usr/local/lib/libqgis_gui.so.3.35.0
#14 0x00007f1a76b1f7bd in QtPrivate::FunctorCall<QtPrivate::IndexesList<0, 1, 2>, QtPrivate::List<QStringList const&, QString const&, QString const&>, void, void (QgsDataSourceManagerDialog::*)(QStringList const&, QString const&, QString const&)>::call(void (QgsDataSourceManagerDialog::*)(QStringList const&, QString const&, QString const&), QgsDataSourceManagerDialog*, void**) () at /usr/local/lib/libqgis_gui.so.3.35.0
#15 0x00007f1a76b1f257 in void QtPrivate::FunctionPointer<void (QgsDataSourceManagerDialog::*)(QStringList const&, QString const&, QString const&)>::call<QtPrivate::List<QStringList const&, QString const&, QString const&>, void>(void (QgsDataSourceManagerDialog::*)(QStringList const&, QString const&, QString const&), QgsDataSourceManagerDialog*, void**) () at /usr/local/lib/libqgis_gui.so.3.35.0
#16 0x00007f1a76b1ec91 in QtPrivate::QSlotObject<void (QgsDataSourceManagerDialog::*)(QStringList const&, QString const&, QString const&), QtPrivate::List<QStringList const&, QString const&, QString const&>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) () at /usr/local/lib/libqgis_gui.so.3.35.0
#17 0x00007f1a73f25442 in  () at /lib64/libQt5Core.so.5
#18 0x00007f1a75c3231d in QgsAbstractDataSourceWidget::addVectorLayers(QStringList const&, QString const&, QString const&) () at /usr/local/lib/libqgis_gui.so.3.35.0
#19 0x00007f1a765ec2ea in QgsOgrSourceSelect::addButtonClicked() () at /usr/local/lib/libqgis_gui.so.3.35.0
#20 0x00007f1a76b1f981 in QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, void (QgsAbstractDataSourceWidget::*)()>::call(void (QgsAbstractDataSourceWidget::*)(), QgsAbstractDataSourceWidget*, void**) () at /usr/local/lib/libqgis_gui.so.3.35.0
#21 0x00007f1a76b1f326 in void QtPrivate::FunctionPointer<void (QgsAbstractDataSourceWidget::*)()>::call<QtPrivate::List<>, void>(void (QgsAbstractDataSourceWidget::*)(), QgsAbstractDataSourceWidget*, void**) () at /usr/local/lib/libqgis_gui.so.3.35.0
#22 0x00007f1a76b1ef5b in QtPrivate::QSlotObject<void (QgsAbstractDataSourceWidget::*)(), QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) () at /usr/local/lib/libqgis_gui.so.3.35.0
#23 0x00007f1a73f25442 in  () at /lib64/libQt5Core.so.5
#24 0x00007f1a74c96ae2 in QAbstractButton::clicked(bool) () at /lib64/libQt5Widgets.so.5
#25 0x00007f1a74c96d4a in  () at /lib64/libQt5Widgets.so.5
#26 0x00007f1a74c98587 in  () at /lib64/libQt5Widgets.so.5
#27 0x00007f1a74c98797 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) () at /lib64/libQt5Widgets.so.5
#28 0x00007f1a74be6d58 in QWidget::event(QEvent*) () at /lib64/libQt5Widgets.so.5
#29 0x00007f1a74ba519e in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /lib64/libQt5Widgets.so.5
#30 0x00007f1a74bad5cf in QApplication::notify(QObject*, QEvent*) () at /lib64/libQt5Widgets.so.5
#31 0x00007f1a6f4b0269 in QgsApplication::notify(QObject*, QEvent*) () at /usr/local/lib/libqgis_core.so.3.35.0
#32 0x00007f1a73eed198 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib64/libQt5Core.so.5
#33 0x00007f1a74bab92e in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) () at /lib64/libQt5Widgets.so.5
#34 0x00007f1a74bff7ed in  () at /lib64/libQt5Widgets.so.5
#35 0x00007f1a74c02d0f in  () at /lib64/libQt5Widgets.so.5
#36 0x00007f1a74ba519e in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /lib64/libQt5Widgets.so.5
#37 0x00007f1a6f4b0269 in QgsApplication::notify(QObject*, QEvent*) () at /usr/local/lib/libqgis_core.so.3.35.0
#38 0x00007f1a73eed198 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib64/libQt5Core.so.5
#39 0x00007f1a7437d0db in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () at /lib64/libQt5Gui.so.5
#40 0x00007f1a7435038c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt5Gui.so.5
#41 0x00007f1a642f954a in  () at /lib64/libQt5XcbQpa.so.5
#42 0x00007f1a69913f30 in  () at /lib64/libglib-2.0.so.0
#43 0x00007f1a69915b58 in  () at /lib64/libglib-2.0.so.0
#44 0x00007f1a6991620c in g_main_context_iteration () at /lib64/libglib-2.0.so.0
#45 0x00007f1a73f460d6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt5Core.so.5
#46 0x00007f1a73eebc2b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt5Core.so.5
#47 0x00007f1a73ef40c0 in QCoreApplication::exec() () at /lib64/libQt5Core.so.5
#48 0x000000000041554c in main ()
[Inferior 1 (process 4319) detached]
gdb returned 0
Aborted (core dumped)
```

Info on about page:
``
QGIS version | 3.35.0-Master | QGIS code branch | master
-- | -- | -- | --
Qt version | 5.15.11
Python version | 3.11.6
GDAL/OGR version | 3.9.0dev
PROJ version | 9.3.1
EPSG Registry database version | v10.098 (2023-11-24)
GEOS version | 3.12.1-CAPI-1.18.1
SQLite version | 3.44.2
PostgreSQL client version | 12.16
SpatiaLite version | 5.1.0
QWT version | 6.2.0
QScintilla2 version | 2.14.1
OS version | openSUSE Tumbleweed
  |   |   |  
Active Python plugins
db_manager | 0.1.20
processing | 2.12.99
grassprovider | 2.12.99
MetaSearch | 0.3.6

``

### Steps to reproduce the issue

Open QGIS
Open Data Source manager
Select a GPX vector file

### Versions

3.35.0-Master (2023-12-07)

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by dhdeangelis** [comment]
A sample GPX is attached here. 
[test.gpx.zip](https://github.com/qgis/QGIS/files/13610768/test.gpx.zip)
No parameters set, as QGIS crashes just after the dialog for choosing layer appears. The attached GPX file can be converted to other formats (e.g. GPKG) using ogr2ogr and imported to QGIS without problems.

**Comment by agiudiceandrea** [comment]
On my Windows 10 system, using QGIS 3.35.0-Master (tested QGIS code revisions 0c2b032849 or dd13b4bfdc) with GDAL/OGR 3.9.0dev-30ab9e70e5-dirty from OSGeo4W, the issue only occurs importing the GPX file via "Data Source Manager" -> "Vector" dialog window (or via drag&drop the file in QGIS), while it doesn't occur importing the same file via "Data Source Manager" -> "GPS" dialog window.
The issue doesn't occur on the same system using QGIS 3.35.0-Master (QGIS code revisions b0df9e2bb0) with GDAL/OGR 3.8.1 from MingW64 build.

**Comment by dhdeangelis** [comment]
Thanks for those tests @agiudiceandrea . I can confirm that the problem does not occur when importing the GPX file via the GPS dialogue in Data Source Manager.

**Comment by dhdeangelis** [comment]
There has been some work on the OGR GPX driver in the last weeks. But, as noted above, ogr2ogr can convert GPX to other formats and the GPX file can be read by GRASS GIS.

```
> ogrinfo -so test.gpx track_points
INFO: Open of `test.gpx'
      using driver `GPX' successful.

Layer name: track_points
Geometry: Point
Feature Count: 1844
Extent: (18.191477, 59.401256) - (18.208581, 59.405273)
Layer SRS WKT:
GEOGCRS["WGS 84",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433]],
    CS[ellipsoidal,2],
        AXIS["geodetic latitude (Lat)",north,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433]],
        AXIS["geodetic longitude (Lon)",east,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433]],
    ID["EPSG",4326]]
Data axis to CRS axis mapping: 2,1
track_fid: Integer (0.0)
track_seg_id: Integer (0.0)
track_seg_point_id: Integer (0.0)
ele: Real (0.0)
time: DateTime
course: Real (0.0)
speed: Real (0.0)
magvar: Real (0.0)
geoidheight: Real (0.0)
name: String (0.0)
cmt: String (0.0)
desc: String (0.0)
src: String (0.0)
url: String (0.0)
urlname: String (0.0)
sym: String (0.0)
type: String (0.0)
fix: String (0.0)
sat: Integer (0.0)
hdop: Real (0.0)
vdop: Real (0.0)
pdop: Real (0.0)
ageofdgpsdata: Real (0.0)
dgpsid: Integer (0.0)
```

**Comment by rouault** [comment]
> Surely @rouault knows if the issue is in the OGR provider in QGIS or in GDAL/OGR (or in a combination of both).

I'm not sure. Likely in GDAL since there have been indeed changes in the GPX driver in master recently (I've reviewed them again, and nothing jumps to my eyes), but GDAL CI works fine on it, including on Windows. Valgrind is happy with ogrinfo or ogr2ogr on the provided file. Debugging QGIS+GDAL on Windows is unfortunately not practical for me.

**Comment by rouault** [comment]
> The issue also occurs on openSUSE, according to the original report.

ah, missed that. Reproduced and fixed in https://github.com/OSGeo/gdal/pull/8941

**Comment by dhdeangelis** [comment]
Thanks @rouault and @agiudiceandrea. I can now confirm this has been solved after the GDAL fix. I am left wondering how such a small thing could trigger such a hard crash in QGIS.

**Comment by rouault** [comment]
> I am left wondering how such a small thing could trigger such a hard crash in QGIS.

it was "just" a null pointer dereference, and that's a hard crash in C/C++ software, and there's no way of recovering from this. It was due to a particular setting that is not the nominal one of the GPX driver, but QGIS sets this option that wasn't tested in the GDAL regression test suite (I've added it now). "What is not tested is broken (or bound to be broken in the future)", my always true motto.
